### PR TITLE
Make brave extension background page conditional

### PR DESCRIPTION
**Pull request by atuchin-m**
For https://github.com/brave/brave-browser/issues/39610 

The PR stops spawning a background page (=an extra renderer by default).
If a user enables WDP or Rewards then background page will be launched.

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
https://github.com/brave/brave-browser/issues/39610#issuecomment-2352471054

### CODE REVIEW

**Review by zenparsing — APPROVED** [review]
Rewards 👍

**Comment by zenparsing** [comment]
Rewards testing:

* Open the browser with a clean profile
* Enable rewards
* Open a new tab and navigate to a page that uses Greaselion (e.g. `https://x.com/bravelaurenwags` on staging)
* Verify that the creator checkmark appears on the Rewards icon
* Reset rewards
* Open a new tab and navigate to a page that uses Greaselion (e.g. `https://x.com/bravelaurenwags` on staging)
* Verify that the creator checkmark does not appear on the Rewards icon

**Review by goodov — APPROVED** [review]

**Review by remusao — APPROVED** [review]

**Review by goodov — APPROVED** [review]

File: browser/extensions/brave_component_loader.cc [hunk 1]
```diff
@@ -153,4 +161,49 @@ void BraveComponentLoader::AddWebTorrentExtension() {
...
+    return;
+  }
+
+  base::FilePath brave_extension_path(FILE_PATH_LITERAL(""));
+  brave_extension_path =
+      brave_extension_path.Append(FILE_PATH_LITERAL("brave_extension"));
+  auto& resource_bundle = ui::ResourceBundle::GetSharedInstance();
+  std::optional<base::Value::Dict> manifest = base::JSONReader::ReadDict(
+      resource_bundle.GetRawDataResource(IDR_BRAVE_EXTENSION));
+  CHECK(manifest) << "invalid Brave Extension manifest";
```
Comment (cdesouza-chromium) [inline browser/extensions/brave_component_loader.cc:183]: There's something going on with this check when building our browser tests in `Debug` mode, where this value returned by the resource bundle call will always be empty. I had seen failures before in Linux, a few weeks ago in fact, but it hadn't occurred to me that those were specific to debug builds.

```
npm run build -- Debug --ignore_compile_failure --target=brave_browser_tests 
../out/Debug/brave_browser_tests --test-launcher-jobs=1 --gtest_filter=BraveContentBrowserClientTest.CanLoadChromeURL
```

@atuchin-m Would you mind taking a look into this?

Comment (cdesouza-chromium) [reply browser/extensions/brave_component_loader.cc:183]: https://github.com/brave/brave-browser/issues/41604

Comment (atuchin-m) [reply browser/extensions/brave_component_loader.cc:183]: I will take a look. The current hypothesis is that the resource bundle isn't property initialized in Debug browser tests.

**Pull request merged**